sinks/signalfx/signalfx.go

```diff
@@ -208,69 +211,102 @@ func (sfx *SignalFxSink) Flush(ctx context.Context, interMetrics []samplers.Inte
 	return err
 }
 
-// FlushEventsChecks sends events to SignalFx. It does not support checks.
+// FlushOtherSamples sends events to SignalFx. It does not support checks.
 func (sfx *SignalFxSink) FlushOtherSamples(ctx context.Context, samples []ssf.SSFSample) {
 	span, _ := trace.StartSpanFromContext(ctx, "")
 	defer span.ClientFinish(sfx.traceClient)
 
+	udpMetrics := make([]samplers.UDPMetric, 0, len(samples))
 	for _, sample := range samples {
 
-		if _, ok := sample.Tags[dogstatsd.EventIdentifierKey]; !ok {
-			// This isn't an event, just continue
-			continue
-		}
-
-		dims := map[string]string{}
+		if _, ok := sample.Tags[dogstatsd.CheckIdentifierKey]; ok {
```

Comment:
Also, I think these two if's would look much better with a switch and a discriminating function (since they're mutually exclusive cases & the two ifs don't quite represent that):

``` go
type ddSampleKind int
const (
    ddSampleEvent ddSampleKind = 0,
    ddSampleServiceCheck = iota,
    ddSampleUnknown = iota,
)

func ddOtherSampleKind(sample *ssf.SSFSample) ddSampleKind {
    if _, ok := sample.Tags[dogstatsd.CheckIdentifierKey]; ok {
        return ddSampleServiceCheck
    }
    if _, ok := sample.Tags[dogstatsd. EventIdentifierKey]; ok {
        return ddSampleEvent
    }
    return ddSampleUnknown
}
```
Then the ifs become:

``` go 
switch ddOtherSampleKind(sample) {
case ddSampleEvent:
    evt := sfx.reportEvent(sample)
    // report evt
case ddSampleServiceCheck
    svchk := sfx.translateServiceCheck(sample)
    // report svchk
default:
    // Report an error metric I guess, and skip it:
    continue;
}
```